src/main/resources/common-services/CDAP/4.0.0/configuration/cdap-site.xml

```diff
@@ -341,6 +341,16 @@
   </property>
 
   <property>
+    <name>app.program.spark.compat</name>
+    <value>spark1_2.10</value>
+    <description>
+      The Spark compatibility version, if Spark is available. Must be either 'spark1_2.10'
+      or 'spark2_2.11'. In distributed mode, this will be overridden by the detected Spark
+      version. In standalone mode, this can be set to the desired version of Spark.
+    </description>
+  </property>
+
+  <property>
     <name>app.program.jvm.opts</name>
     <value>-XX:MaxPermSize=128M ${twill.jvm.gc.opts} -Dhdp.version=${hdp.version} -Dspark.yarn.am.extraJavaOptions=-Dhdp.version=${hdp.version}</value>
```

Comment:
Remove the `-Dspark.yarn.am.extraJavaOptions` from this due to caskdata/cdap_cookbook#243